### ISSUE TITLE
Build Network Topology Table throws error with HexSim stream network

### ISSUE THREAD

**Issue by jesselangdon** [SouthForkResearch/gnat#25]
```
....
  Single Reach, 6543
  Single Reach, 6549
Traceback (most recent call last):
  File "<string>", line 488, in execute
  File "C:\dev\gnat\BuildNetworkTopology.py", line 319, in main
    network_tree(downstream_oid,tableNetwork,fcStreamNetworkTemp_lyr,fcNodePoint)
(...multiple calls at line 121...) 
  File "C:\dev\gnat\BuildNetworkTopology.py", line 121, in network_tree
    network_tree(listSelected[0],tblNetwork,fcLines,fcNodePoint)
  File "C:\dev\gnat\BuildNetworkTopology.py", line 58, in network_tree
    arcpy.Delete_management("InputReach")
  File "c:\program files (x86)\arcgis\desktop10.1\arcpy\arcpy\management.py", line 3658, in Delete
    raise e
RuntimeError: maximum recursion depth exceeded while calling a Python object

Failed to execute (BuildNetworkTopologyTool).
Failed at Mon Apr 17 14:29:43 2017 (Elapsed Time: 8 minutes 56 seconds)
```

**Comment by KellyMWhitehead** [comment]
Might be related to this: 
https://stackoverflow.com/questions/3323001/what-is-the-maximum-recursion-depth-in-python-and-how-to-increase-it

**Comment by jesselangdon** [comment]
I did increase the Python recursion limit to 2000 (`sys.setrecursionlimit(2000)`), which allowed the process to finish in an IDE debugging session.  What's odd is that Python we encounter a recursion limit with this particular dataset, because it's worked on larger stream network datasets before (i.e. Lower John Day), without too many problems.  

Increasing the recursion limit manually is not recommended however (see the link above), as it can result in a stack overflow.  So I don't think this is a long term solution...  ultimately the code should be reworked so that we use a loop instead of recursion, but that would take a pretty big refactoring effort.  I agree with Kelly that refactoring efforts should be punted for now... the new QGIS plugin uses a totally different technique for dealing with topology.  In the meantime, I've added an increase to the recursion limit, when this process runs.
